bot/cogs/grades_data/grades_cog.py

```diff
@@ -234,6 +242,59 @@ def searchCourse(self, query, year):
     def go(self, course, year):
         return self.searchCourse(course, year)
 
+    def get_professor_query(self, prof_name, detailed):
+        # print("NAME: " + prof_name)
+        primary = self.global_master_prof_list[prof_name]
+        courses = {}
+        
+        for section in primary:
+            course = section['course'] + "-" + section['number']
+            courses[course] = {
```

Comment:
this could be more idiomatically represented as a dataclass, cut down on dict string indexing